ksql-execution/src/main/java/io/confluent/ksql/execution/function/udtf/KudtfFlatMapper.java

```diff
@@ -28,20 +29,32 @@
 @Immutable
 public class KudtfFlatMapper implements ValueMapper<GenericRow, Iterable<GenericRow>> {
 
-  private final TableFunctionApplier functionHolder;
+  private final List<TableFunctionApplier> tableFunctionAppliers;
 
-  public KudtfFlatMapper(final TableFunctionApplier functionHolder) {
-    this.functionHolder = Objects.requireNonNull(functionHolder);
+  public KudtfFlatMapper(final List<TableFunctionApplier> tableFunctionAppliers) {
+    this.tableFunctionAppliers = Objects.requireNonNull(tableFunctionAppliers);
   }
 
   @Override
   public Iterable<GenericRow> apply(final GenericRow row) {
-    final List<Object> exploded = functionHolder.apply(row);
-    final List<GenericRow> rows = new ArrayList<>(exploded.size());
-    for (Object val : exploded) {
-      final List<Object> arrayList = new ArrayList<>(row.getColumns());
-      arrayList.add(val);
-      rows.add(new GenericRow(arrayList));
+    final List<Iterator<Object>> iters = new ArrayList<>(tableFunctionAppliers.size());
+    int maxLength = 0;
+    for (TableFunctionApplier applier: tableFunctionAppliers) {
+      final List<Object> exploded = applier.apply(row);
+      iters.add(exploded.iterator());
+      maxLength = Math.max(maxLength, exploded.size());
+    }
+    final List<GenericRow> rows = new ArrayList<>(maxLength);
+    for (int i = 0; i < maxLength; i++) {
```

Comment:
might be good to have an inline comment describing the zipjoin behavior of multiple UDTFs to set expectations reading this code (or point to some documentation about how multiple table functions work)